corehq/apps/geospatial/static/geospatial/js/geo_config.js

```diff
@@ -46,6 +50,33 @@ hqDefine("geospatial/js/geo_config", [
             return self.selectedGroupMethod() === minMaxGroupingName;
         });
 
+        self.selectedAlgorithm = ko.observable();
+        self.apiToken = ko.observable(data.api_token);
+
+        self.captureApiToken = ko.computed(function () {
+            return self.selectedAlgorithm() === ROAD_NETWORK_ALGORITHM;
+        });
+
+        self.validateApiToken = function () {
+            const url = "https://api.mapbox.com/directions-matrix/v1/mapbox/driving/-122.42,37.78;-122.45,37.91;-122.48,37.73";
+            const params = "access_token=" + self.apiToken();
+
+            $.ajax({
+                method: 'get',
+                url: url + "?" + params,
+                success: function (data, status, request) {
+                    if (request.status === 200) {
```

Comment:
According to the [docs](https://docs.mapbox.com/api/navigation/matrix/#matrix-api-errors) one will always receive either a 200 or >= 400 status codes, in which case the `fail` subscription block will trigger.

Maybe I should collapse the success block if? I do concur that if there's something that happens that is not a 200 status code but triggered the `success` flow it could be weird...but then again, theoretically it won't happen.

Reply:
Since we know what to expect from Mapbox, I think it should be fine to completely remove this conditional block. I agree that it would be quite strange, and very unlikely, to receive something other than 200 for success.